File: backend/routers/bookmark.py
# backend/routers/bookmark.py

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from backend import models, schemas
from backend.db import get_db
from backend.app.auth.dependencies import get_current_user

router = APIRouter(
    prefix="/bookmarks",
    tags=["Bookmark"]
)

@router.post("/", response_model=schemas.BookmarkOut)
def create_bookmark(
    bookmark: schemas.BookmarkCreate,
    db: Session = Depends(get_db),
    current_user: models.User = Depends(get_current_user)
):
    existing_bookmark = db.query(models.Bookmark).filter_by(
        user_id=current_user.id,
        recipe_id=bookmark.recipe_id
    ).first()

    if existing_bookmark:
        raise HTTPException(status_code=400, detail="Already bookmarked this recipe.")

    db_bookmark = models.Bookmark(user_id=current_user.id, recipe_id=bookmark.recipe_id)
    db.add(db_bookmark)
    db.commit()
    db.refresh(db_bookmark)
    return db_bookmark

@router.get("/me", response_model=list[schemas.BookmarkOut])
def get_my_bookmarks(
    db: Session = Depends(get_db),
    current_user: models.User = Depends(get_current_user)
):
    return db.query(models.Bookmark).filter(models.Bookmark.user_id == current_user.id).all()

@router.delete("/{bookmark_id}", status_code=204)
def delete_bookmark(
    bookmark_id: int,
    db: Session = Depends(get_db),
    current_user: models.User = Depends(get_current_user)
):
    db_bookmark = db.query(models.Bookmark).filter(
        models.Bookmark.id == bookmark_id,
        models.Bookmark.user_id == current_user.id
    ).first()
    if not db_bookmark:
        raise HTTPException(status_code=404, detail="Bookmark not found")
    db.delete(db_bookmark)
    db.commit()
    return

File: frontend-web/src/api/bookmark.js
import client from './client';
import { BOOKMARK } from './endpoints';

/**
 * 북마크 추가 API
 * @param {Object} bookmarkData - 북마크 정보
 * @param {number} bookmarkData.recipe_id - 레시피 ID
 * @returns {Promise} 생성된 북마크 정보
 */
export const createBookmark = async (bookmarkData) => {
  try {
    const response = await client.post(BOOKMARK.CREATE, bookmarkData);
    return response.data;
  } catch (error) {
    console.error('Create bookmark error:', error);
    throw error;
  }
};

/**
 * 내 북마크 목록 조회 API
 * @returns {Promise} 사용자의 북마크 목록
 */
export const getMyBookmarks = async () => {
  try {
    const response = await client.get(BOOKMARK.LIST_ME);
    return response.data;
  } catch (error) {
    console.error('Get my bookmarks error:', error);
    throw error;
  }
};

/**
 * 북마크 삭제 API
 * @param {number} bookmarkId - 북마크 ID
 * @returns {Promise} 삭제 결과
 */
export const deleteBookmark = async (bookmarkId) => {
  try {
    const response = await client.delete(BOOKMARK.DELETE(bookmarkId));
    return response.data;
  } catch (error) {
    console.error(`Delete bookmark ${bookmarkId} error:`, error);
    throw error;
  }
};